From b729d9abe4e347cdd7058da31055fb36465a8745 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Thu, 21 Mar 2024 08:32:29 +0100
Subject: [PATCH] arm64: dts: st: set Tx and Rx Qos of Ethernet1 and Ethernet2
 on stm32mp25

All QoS IP priorities are managed through SYSCFG_ICNQPCR register.
Except Ethernet QoS which is managed inside a dedicated
register (ETH_DMACxRXCR) in Ethernet IP.
The QoS value is currently 0 (hardware default value),
which is the problem.
Bandwidth regulator is enabled. When DDR bandwidth
reaches a certain level, low QoS priorities have their bandwidth
even more decreased.
As ethernet QoS is configured to 0, this is the first one impacted.
Configure the Ethernet QoS to 7.

Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
Change-Id: I7317dc549ac288fc1131292987fa1555bfde54e7
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368139
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/410118
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 2 ++
 arch/arm64/boot/dts/st/stm32mp253.dtsi | 2 ++
 2 files changed, 4 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2299,6 +2299,8 @@
 				st,syscon = <&syscfg 0x3000 0xffffffff>;
 				snps,mixed-burst;
 				snps,pbl = <2>;
+				snps,txqos = <7>;
+				snps,rxqos = <7>;
 				snps,axi-config = <&stmmac_axi_config_1>;
 				snps,tso;
 				access-controllers = <&rifsc 60>;
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -168,6 +168,8 @@
 		st,syscon = <&syscfg 0x3400 0xffffffff>;
 		snps,mixed-burst;
 		snps,pbl = <2>;
+		snps,txqos = <7>;
+		snps,rxqos = <7>;
 		snps,axi-config = <&stmmac_axi_config_2>;
 		snps,tso;
 		access-controllers = <&rifsc 61>;
